tests/unit/test_extensionPoints.py

```diff
@@ -0,0 +1,255 @@
+#tests/unit/test_extensionPoints.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""Unit tests for the extensionPoints module.
+"""
+
+import unittest
+import extensionPoints
+
+class ExampleClass(object):
+	def method(self):
+		return 42
+
+def exampleFunc():
+	return 3.14
+
+class TestBoundMethodWeakref(unittest.TestCase):
+
+	def onDelete(self, weak):
+		self.deleted = weak
+
+	def setUp(self):
+		self.deleted = None
+		self.instance = ExampleClass()
+		self.weak = extensionPoints.BoundMethodWeakref(self.instance.method, self.onDelete)
+
+	def test_get(self):
+		"""Test that we get the right strong reference from the weak reference.
+		"""
+		method = self.weak()
+		self.assertEqual(method, self.instance.method)
+
+	def test_onDelete(self):
+		"""Test that the deletion callback gets called.
+		"""
+		del self.instance
+		self.assertEqual(self.deleted, self.weak)
+
+	def test_isWeak(self):
+		"""Test that this is actually a weak reference;
+		i.e. that it dies when the instance dies.
+		"""
+		del self.instance
+		method = self.weak()
+		self.assertIsNone(method)
+
+class TestCallWithSupportedKwargs(unittest.TestCase):
+
+	def test_supportsNoKwargs(self):
+		called = []
+		def h():
+			called.append(True)
+		extensionPoints.callWithSupportedKwargs(h, a=1)
+		self.assertEqual(called, [True])
+
+	def test_supportsLessKwargs(self):
+		gotKwargs = {}
+		def h(a=None):
+			gotKwargs["a"] = a
+		extensionPoints.callWithSupportedKwargs(h, a=1, b=2)
+		self.assertEqual(gotKwargs, {"a": 1})
+
+	def test_supportsExtraKwargs(self):
+		gotKwargs = {}
+		def h(a=None, b=2):
+			gotKwargs["a"] = a
+			gotKwargs["b"] = b
+		extensionPoints.callWithSupportedKwargs(h, a=1)
+		self.assertEqual(gotKwargs, {"a": 1, "b": 2})
+
+	def test_supportsAllKwargs(self):
+		gotKwargs = {}
+		def h(**kwargs):
+			gotKwargs.update(kwargs)
+		extensionPoints.callWithSupportedKwargs(h, a=1)
+		self.assertEqual(gotKwargs, {"a": 1})
+

```

Comment:
> what about where no defaults are provided, the same cases as `test_supportsLessKwargs` and `test_supportsExtraKwargs` but with `def h(a, b)`

Those would be positional arguments, which are passed completely unmodified, since this function is only supposed to mess with kwargs. Still, I've added a test to at least check that positionals do get passed in case the code gets modified and someone accidentally removes the `*args` bit.